src/Test/Runner/Failure.elm

```diff
@@ -1,7 +1,39 @@
-module Test.Message exposing (failureMessage)
+module Test.Runner.Failure exposing (InvalidReason(..), Reason(..), format)
 
-import String
-import Test.Expectation exposing (InvalidReason(..), Reason(..))
+{-| The reason a test failed.
+
+@docs Reason, InvalidReason, format
+
+-}
+
+
+{-| -}
+type Reason
+    = Custom
+    | Equals String String
+    | Comparison String String
```

Comment:
`Equality` would match `Comparison` better, and better imply that `notEqual` also produces this.

Is it worth having `type Comparison = (four possibilities)`?